skfda/exploratory/visualization/_outliergram.py

```diff
@@ -0,0 +1,152 @@
+"""Outliergram Module.
+
+This module contains the methods used to plot shapes in order to detect
+shape outliers in our dataset. In order to do this, we plot the
+Modified Band Depth and Modified Epigraph Index, that will help us detect
+this outliers. The motivation of the method is that it is easy to find
+magnitude outliers, but there is a necessity of capturing this other type.
+"""
+
+from typing import List, Optional, TypeVar
+
+import scipy.integrate as integrate
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from scipy.stats import rankdata
+
+from ... import FDataGrid
+from ..depth._depth import ModifiedBandDepth
+from ._utils import _get_figure_and_axes, _set_figure_layout_for_fdata
+
+S = TypeVar('S', Figure, Axes, List[Axes])
+
+
+class Outliergram:
+    """
+    Outliergram method of visualization.
+
+    Plots the Modified Band Depth (MBD) on the Y axis and the Modified
+    Epigraph Index (MEI) on the X axis. This points will create the form of
+    a parabola. The shape outliers will be the points that appear far from
+    this curve.
+    Args:
+        fdata: functional data set that we want to examine.
+    Attributes:
+        mbd: result of the calculation of the Modified Band Depth on our
+            dataset. Represents the mean time a curve stays between other pair
+            of curves, being a good measure of centrality.
+        mei: result of the calculation of the Modified Epigraph Index on our
+            dataset. Represents the mean time a curve stays below other curve.
+    References:
+        López-Pintado S.,  Romo J.. (2011). A half-region depth for functional
+        data, Computational Statistics & Data Analysis, volume 55
+        (page 1679-1695).
+        Arribas-Gil A., Romo J.. Shape outlier detection and visualization for
+        functional data: the outliergram
+        https://academic.oup.com/biostatistics/article/15/4/603/266279
+    """
+
+    def __init__(
+        self,
+        fdata: FDataGrid,
+    ) -> None:
+        self.fdata = fdata
+        self.depth = ModifiedBandDepth()
+        self.depth.fit(fdata)
+        self.mbd = self.depth(fdata)
+        self.mei = self.modified_epigraph_index_list()
+
+    def plot(
+        self,
+        chart: Optional[S] = None,
```

Comment:
Why `S`? A `TypeVar` should have more than one use, otherwise use a normal type.

Reply:
true, corrected with a union